### ISSUE TITLE
Make UserNameReference to allow for alternate format database connection strings

### ISSUE THREAD

**Issue by onionhammer** [dotnet/aspire#5459]
This property is internal, so if you need to create a different connection string format for your aspire resources (say psql format), it becomes more difficult

https://github.com/dotnet/aspire/blob/f5d9f80c3c667f5c84debfd427e7b0ae83493f7d/src/Aspire.Hosting.PostgreSQL/PostgresServerResource.cs#L42

**Comment by davidfowl** [comment]
We can just expose this property

**Comment by eerhardt** [comment]
@onionhammer - can you explain the scenario a bit more? The UserNameParameter is exposed, so if you were making your own ReferenceExpression for a connection string, you should be able to use it.

**Comment by onionhammer** [comment]
@eerhardt Yes, but only if the username parameter is specified; what if it's not specified?

**Comment by eerhardt** [comment]
> what if it's not specified?

Then use the default `postgres`.

> I don't see much purpose for it being internal, other than leaving open the possibility for refactors without breaking dependent code

The thinking is to only make things public if there are user scenarios for using them. That way we don't have to go through breaking changes, or leave old APIs in.

Can you explain your scenario more? That would help prioritize this.

**Comment by onionhammer** [comment]
This was back in August, so I think back then I was trying to figure out how to hook up a map tiling server ( https://maplibre.org/martin/env-vars.html ) , and worked with @ReubenBond who said I ought to open an issue for it

![Image](https://github.com/user-attachments/assets/2284c4ce-bd57-4c74-9006-d681dba8d5d0)

Eventually I came up with this

![Image](https://github.com/user-attachments/assets/bcbd6686-77a7-49da-a28b-33723a65063d)

I think the goal is to make it simpler to build connection strings without duplicating the username / magic string default username logic in multiple places
